packages/api/src/store/stream-table.ts

```diff
@@ -1,25 +1,158 @@
+    res = await this.db.queryWithOpts(q1, opts);
+
+    if (res.rowCount > 0) {
+      for (const row of res.rows) {
+        usage.push({
+          id: row.id,
+          date: row.date,
+          sourceSegments: row.sourcesegments,
+          transcodedSegments: row.transcodedsegments,
+          sourceSegmentsDuration: row.sourcesegmentsduration,
+          transcodedSegmentsDuration: row.transcodedsegmentsduration,
+          streamCount: row.streamcount,
+        });
+      }
+    }
+    return usage;
+  }
+
+  async usageHistory(fromTime: number, toTime: number, opts?: QueryOptions) {
+    const q1 = sql`SELECT
+      TO_TIMESTAMP((data->>'createdAt')::bigint/1000)::date as day,
+      sum((data->>'sourceSegmentsDuration')::float) as sourceSegmentsDuration,
+      sum((data->>'transcodedSegmentsDuration')::float) as transcodedSegmentsDuration,
+      sum((data->>'transcodedSegments')::int) as transcodedSegments, 
+      sum((data->>'sourceSegments')::numeric) as sourceSegments,
+      count(*)::int as streamCount
+      
+    FROM stream WHERE data->>'sourceSegmentsDuration' IS NOT NULL
+      AND data->>'parentId' IS NOT NULL 
+      AND (data->>'createdAt')::bigint >= ${fromTime}
+      AND (data->>'createdAt')::bigint < ${toTime}
+      GROUP BY day
+      ORDER BY day
+    `;
+
+    let usage = [];
+
+    let res: QueryResult<dbUsageHistoryData>;
+    res = await this.db.queryWithOpts(q1, opts);
+
+    if (res.rowCount > 0) {
+      for (const row of res.rows) {
+        let dayStartTimestamp = new Date(row.day).setUTCHours(0, 0, 0, 0);
+        usage.push({
+          id: new Date(row.day).toLocaleDateString("en-CA"),
+          date: dayStartTimestamp,
+          sourceSegments: +row.sourcesegments,
+          transcodedSegments: +row.transcodedsegments,
+          sourceSegmentsDuration: row.sourcesegmentsduration,
+          transcodedSegmentsDuration: row.transcodedsegmentsduration,
+          streamCount: row.streamcount,
+        });
+      }
+    }
+    const q2 = sql`SELECT
+      TO_TIMESTAMP((data->>'createdAt')::bigint/1000)::date as day,
+      sum((data->>'sourceSegmentsDuration')::float) as sourceSegmentsDuration,
+      sum((data->>'transcodedSegmentsDuration')::float) as transcodedSegmentsDuration,
+      sum((data->>'transcodedSegments')::int) as transcodedSegments, 
+      sum((data->>'sourceSegments')::numeric) as sourceSegments,
+      count(*)::int as streamCount
+      
+    FROM stream s WHERE data->>'sourceSegmentsDuration' IS NOT NULL
+      AND data->>'parentId' IS NULL
+      AND (SELECT COUNT(C.id) FROM stream C WHERE C.data->>'parentId' = S.Id) = 0
+      AND (data->>'createdAt')::bigint >= ${fromTime}
+      AND (data->>'createdAt')::bigint < ${toTime}
+      GROUP BY day
+      ORDER BY day
+    `;
+    res = await this.db.queryWithOpts(q2, opts);
+
+    if (res.rowCount > 0) {
+      for (const row of res.rows) {
+        let existingRow = usage.filter((r) => r.day === row.day);
```

Comment:
This seems like it's going to end up being a pretty expensive operation. If there are N rows for query q1 and M rows for query q2, we'll then have M for loop iterations and in each of those loop iterations we'll have N iterations for the filter function (assuming that the filter function works something like that).

Could this be avoided by using a map to track known days when we create the usage array based on the results of query q1 and then have a single for loop here that checks if the ID of one of these rows is in the map with known IDs? Ex.

```javascript
res = await this.db.queryWithOpts(q1, opts)

knownDays = {}
if (res.rowCount > 0) {
    for (const row of res.rows) {
        ...
        knownDays[row.day] = ... // index of usage object in usage array
    }
}

...
res = await this.db.queryWithOpts(q2, opts)

if (res.rowCount > 0) {
    for (const row of res.rows) {
        if (knownDays[row.day]) {
            // Add usage data to existing usage object
        } else {
            // Create new usage object
        }
    }
}
```